src/layout.js

```diff
@@ -208,9 +208,12 @@ export function getLengthUnits(length) {
 /**
  * Returns the numeric value of a CSS length value.
  * @param {!LengthDef|string} length
- * @return {number}
+ * @return {number|undefined}
  */
 export function getLengthNumeral(length) {
+  if (!length) {
```

Comment:
I just saw another thread going to fixing `Number.is***` elsewhere. So let's just focus on his use case.

@zhouyx `100%` is currently not allowed here. If we do - it'd still be fine. It'd just parse into `100.0`.

@sriramkrish85 We can certainly prohibit `auto` from here, but there's a lot of code already calling this method, so this is the best fix at this time.